File: 0102_Binary_Tree_Level_Order_Traversal/Code_Zero.java
/**

Exam:
Given binary tree [3,9,20,null,null,15,7],
    3
   / \
  9  20
    /  \
   15   7
return its level order traversal as:
[
  [3],
  [9,20],
  [15,7]
]
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new LinkedList<List<Integer>>();
        if (root == null) 
            return result;
        LinkedList<TreeNode> queue = new LinkedList<TreeNode>();
            queue.offer(root);
        while(!queue.isEmpty()) {
            int size = queue.size();
            LinkedList<Integer> list = new LinkedList<Integer>();
            for (int i = 0; i < size; ++i) {
                TreeNode node = queue.pop();
                list.add(node.val);
                if (node.left != null) 
                    queue.offer(node.left);
                if (node.right != null) 
                    queue.offer(node.right);
            }
            result.add(list);
        }
        return result;
    }
}

File: 0101_Symmetric_Tree/Code_Zero.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        def isSameTree(biTree1, biTree2):
            # 若子树皆为空，则返回True
            if not biTree1 and not biTree2:
                return True
            if biTree1 and biTree2 and biTree1.val == biTree2.val:
                # 递归操作
                leftTree = isSameTree(biTree1.left, biTree2.right)
                rightTree = isSameTree(biTree1.right, biTree2.left)
                return leftTree and rightTree
            else:
                return False
        if not root:
            return True
        else:
            return isSameTree(root.left, root.right)
        
